### ISSUE TITLE
Build with `-fno-strict-aliasing` by default

### ISSUE THREAD

**Issue by DemiMarie** [openssl/openssl#18315]
To avoid bugs similar to #18225, OpenSSL should build with `-fno-strict-aliasing` by default.

**Comment by paulidale** [comment]
Do you have some performance measurement to see what's lost?

**Comment by t8m** [comment]
IMO this is bad idea. Instead the code should be reviewed for strict aliasing violations and fixed.

**Comment by DemiMarie** [comment]
> IMO this is bad idea. Instead the code should be reviewed for strict aliasing violations and fixed.

It should indeed be, but given that such bugs are very difficult to diagnose and impact security-critical code, it is better to use `-fno-strict-aliasing` by default for safety in release builds.

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input

**Comment by thesamesam** [comment]
My position per https://github.com/openssl/openssl/issues/18663#issuecomment-1842789985 remains the same.

**Comment by nhorman** [comment]
If no one has worked on it for over two  years, then its somewhat evident that no one has the time to invest in it, and its only purpose here is to clutter the backlog of issues, making work that we do want to concentrate on harder to find.

The path forward here then is to mark this as a community issue, and look for an community developer to fix it.  If one of you is volunteering to do so, I am happy to mark it as such.

**Comment by DemiMarie** [comment]
@nhorman where would one add `-fno-strict-aliasing` to OpenSSL’s build system?

**Comment by nhorman** [comment]
well, it can be done already as part as part of the configuration step:
./Configure <options> -fno-strict-aliasing

that will build with the option you want.

if you want to incorporate it into existing targets automatically, then you need to go poking about in the Configurations directory, there you will find all of our targets listed, with the directives to include various compiler flags for certain targets.  The goal would be to identify the highest leaf in the tree of the apropriate file where the command could be inserted, and not break other platform (which might necessitate some fan out and repitition to specific targets).  Start by looking at 10-main.conf

Can I assume from your question that you're interesting in taking this on as a community issue? :)  I'd be greatful.

**Comment by thesamesam** [comment]
Yes, we're happy to handle it as a community issue. I actually offered a while ago in https://github.com/openssl/openssl/issues/18663#issuecomment-1842789985 ;)

**Comment by nhorman** [comment]
horray!  And thank you!

I'm marking this as a community effort and assigning @DemiMarie 

Just for future reference (and I'm taking on an effort to try communicate this much more clearly), Anyone is perfectly welcome to take on any issue in our list and propose a  PR for it.  If you're looking for a more constrained set of issues, please filter on the help wanted label.  Those issues are/should be valid, but are identified as issues that the core development team doesn't have capacity to consider for a specific release, which is not to say they can't go in, just that we don't have time for them at the moment.  Please, PLEASE feel free to take any  of those issue, assign yourself and raise PRs for them, we are very grateful for the contribution!

**Comment by DemiMarie** [comment]
@thesamesam would you be willing to take this?  I’m afraid I can’t test on anything but recent GCC and Clang, for which the trivial solution is “pass `-fno-strict-aliasing` unconditionally”.

**Comment by thesamesam** [comment]
Go on then. I'm quite busy at the moment but I will handle it.

**Comment by nhorman** [comment]
done, and thank you!

**Comment by t8m** [comment]
I am closing this as this is not going to be done by the team, and it had help wanted label for extended period of time.
